exp/api/v1beta1/azuremachinepool_webhook.go

```diff
@@ -319,3 +322,45 @@ func (amp *AzureMachinePool) ValidateOrchestrationMode(c client.Client) func() e
 		return nil
 	}
 }
+
+// ValidateInstanceMix validates certain instance mix limitations are not violated
+func (amp *AzureMachinePool) ValidateInstanceMix() error {
+	const standardPrefix = "Standard_"
+	supportedFamilies := []string{"A", "B", "D", "E", "F"}
+
+	// Only up to five VM sizes can be specified.
+	if amp.Spec.OrchestrationMode == infrav1.FlexibleOrchestrationMode && len(amp.Spec.Template.AdditionalVMSizes) > 4 {
+		return errors.New("a maximum of 4 additionalVmSizes is allowed when using flexible orchestration")
+	}
+
+	// Check that all VM sizes are from supported families
+	for _, vmSize := range amp.Spec.Template.AdditionalVMSizes {
+		// Extract the VM family from the VM size
+		// Azure VM sizes follow the pattern: Standard_{Family}{Size}[qualifiers]
+		// Examples: Standard_D4s_v3 -> D, Standard_B2ms -> B
+		if !strings.HasPrefix(vmSize.Size, standardPrefix) {
+			return errors.New(fmt.Sprintf("vm size in additionalVmSizes is expected to start with %s, got %s", standardPrefix, vmSize.Size))
+		}
+
+		// Remove the "Standard_" prefix
+		remainder := vmSize.Size[len(standardPrefix):]
+		if len(remainder) == 0 {
+			return errors.New("vm size in additionalVmSizes is expected to have a family and size, got nothing")
+		}
+
+		// The first character after "Standard_" should be the family letter for supported families
+		family := string(remainder[0])
+
+		// Validate that it's actually a letter A-Z
+		if family < "A" || family > "Z" {
+			return errors.New(fmt.Sprintf("unexpected format for VM size, expected letter for family, got %s", family))
+		}
```

Comment:
I think this check redundant as the `slice.Contains` above will return an error if the family is not present in the supportedFamilies slice.